src/main/java/seedu/address/model/person/Person.java

```diff
@@ -104,15 +113,30 @@ public Person withPicture(Picture picture) {
     }
 
     public Person withDates(List<Event> dates) {
-        return new Person(name, phone, email, birthday, address, picture, tags, dates, meetings);
+        return new Person(name, phone, email, birthday, goal, address, picture, tags, dates, meetings);
+    }
+
+    public Person withGoal(Goal goal) {
+        return new Person(name, phone, email, birthday, goal, address, picture, tags, dates, meetings);
     }
 
     public List<Event> getMeetings() {
         return Collections.unmodifiableList(meetings);
     }
 
     public Person withMeetings(List<Event> meetings) {
-        return new Person(name, phone, email, birthday, address, picture, tags, dates, meetings);
+        return new Person(name, phone, email, birthday, goal, address, picture, tags, dates, meetings);
+    }
+
+    public LocalDate getGoalDeadline(LocalDate date) {
+        if (this.meetings.isEmpty()) {
```

Comment:
Need follow the style guide, don't use `this` in instance methods if there is no need to. Link [here](https://se-education.org/guides/conventions/java/index.html#variables)